roles/openshift_health_checker/openshift_checks/package_version.py

```diff
@@ -17,9 +30,86 @@ def is_active(cls, task_vars):
+                    False
+                ),
+                (
+                    "docker",
+                    self.get_required_docker_version(task_vars),
+                    False
+                ),
+                (
+                    "{}".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+                (
+                    "{}-master".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+                (
+                    "{}-node".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+            ],
         }
+
         return self.execute_module("aos_version", args, tmp, task_vars)
+
+    def get_required_ovs_version(self, task_vars):
+        """Return the correct Open vSwitch version for the current OpenShift version.
+        If the current OpenShift version is >= 3.5, ensure Open vSwitch version 2.6,
+        Else ensure Open vSwitch version 2.4"""
+        openshift_version = self.get_openshift_version(task_vars)
+
+        if openshift_version < 3.5:
+            return self.openshift_to_ovs_version["3.4"]
+
+        if not self.openshift_to_ovs_version.get(str(openshift_version)):
+            msg = "There is no recommended version of Open vSwitch for the current version of OpenShift: {}"
+            raise OpenShiftCheckException(msg.format(openshift_version))
+
+        return self.openshift_to_ovs_version[str(openshift_version)]
+
+    def get_required_docker_version(self, task_vars):
+        """Return the correct Docker version for the current OpenShift version.
+        If the OpenShift version is 3.1, ensure Docker version 1.8.
+        If the OpenShift version is 3.2 or 3.3, ensure Docker version 1.10.
+        If the current OpenShift version is >= 3.4, ensure Docker version 1.12."""
+        openshift_version = self.get_openshift_version(task_vars)
+
+        if openshift_version >= 3.4:
+            return self.openshift_to_docker_version["3.4"]
+
+        if not self.openshift_to_docker_version.get(str(openshift_version)):
+            msg = "There is no recommended version of Docker for the current version of OpenShift: {}"
+            raise OpenShiftCheckException(msg.format(openshift_version))
+
+        return self.openshift_to_docker_version[str(openshift_version)]
+
+    def get_openshift_version(self, task_vars):
+        openshift_version = get_var(task_vars, "openshift_image_tag")
+        if openshift_version and openshift_version[0] == 'v':
+            openshift_version = openshift_version[1:]
+
+        return self.parse_version(openshift_version)
+
+    @staticmethod
+    def parse_version(version):
+        components = version.split(".")
+        if not components or len(components) < 2:
+            msg = "Invalid version format returned by host: {}"
```

Comment:
I don't see how a user would ever see this in practice, but if they did, wouldn't this message be confusing, saying the host returned a version? Maybe: "Invalid OpenShift version found for this host:"